src/core/backbone_router/leader.hpp

```diff
@@ -68,6 +69,15 @@ class Leader : public InstanceLocator
         kStateUnchanged,      // No change on Primary Backbone Router information (only for logging).
     };
 
+    enum DomainPrefixState
+    {
+        kDomainPrefixNone = 0,  ///< No Domain Prefix.
+        kDomainPrefixAdded,     // Newly added.
+        kDomainPrefixRemoved,   // Newly removed.
+        kDomainPrefixRefreshed, // Domain Prefix changed.
+        kDomainPrefixUnchanged, // No change on Domain Prefix.
+    };
```

Comment:
Can we define it as `private` in this class?

I think we only use it in this class and for logging purpose. How about calling it `DomainPrefixChange` as it tracks the last change to domain prefix and not a current state.

Reply:
Along with the change from previous notifier way to direct call way, now other components need to access this enumeration.  So keep it public here.